I can not compile Ethernet.h on last - MegaCoreX v1.1.3

My arduino do not compile when included Ethernet.h library with MegaCoreX v1.1.3
There is no problem with old versions.
Please check it. Do i have a mistake?

My test code is:

```
#include <Ethernet.h>

void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}

```

the entire output is:



```
FQBN: MegaCoreX:megaavr:4808
Using board '4808' from platform in folder: C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3
Using core 'coreX-corefiles' from platform in folder: C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3

Detecting libraries used...
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard C:\Users\user\AppData\Local\Temp\arduino\sketches\62E2E3EFDD73DEE05A85442B7B18B5D6\sketch\sketch_nov17a.ino.cpp -o nul
Alternatives for Ethernet.h: [Ethernet@2.0.1 Ethernet@2.0.2]
ResolveLibrary(Ethernet.h)
  -> candidates: [Ethernet@2.0.1 Ethernet@2.0.2]
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src C:\Users\user\AppData\Local\Temp\arduino\sketches\62E2E3EFDD73DEE05A85442B7B18B5D6\sketch\sketch_nov17a.ino.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dhcp.cpp -o nul
Alternatives for SPI.h: [SPI@1.0]
ResolveLibrary(SPI.h)
  -> candidates: [SPI@1.0]
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dhcp.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\EthernetClient.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\EthernetServer.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\EthernetUdp.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\socket.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src c:\Users\user\Documents\Arduino\libraries\Ethernet\src\utility\w5100.cpp -o nul
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src\SPI.cpp -o nul
Generating function prototypes...
C:\Users\user\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/deprecated -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\variants\32pin-standard -Ic:\Users\user\Documents\Arduino\libraries\Ethernet\src -IC:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI\src C:\Users\user\AppData\Local\Temp\arduino\sketches\62E2E3EFDD73DEE05A85442B7B18B5D6\sketch\sketch_nov17a.ino.cpp -o C:\Users\user\AppData\Local\Temp\3241236753\sketch_merged.cpp
C:\Users\user\AppData\Local\Arduino15\packages\builtin\tools\ctags\5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives C:\Users\user\AppData\Local\Temp\3241236753\sketch_merged.cpp
Compiling sketch...
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\sketch\\sketch_nov17a.ino.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\sketch\\sketch_nov17a.ino.cpp.o"
Compiling libraries...
Compiling library "Ethernet"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\Dhcp.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\Dhcp.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\Ethernet.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\Ethernet.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\EthernetClient.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\EthernetClient.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\Dns.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\Dns.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\socket.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\socket.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\EthernetServer.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\EthernetServer.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\EthernetUdp.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\EthernetUdp.cpp.o"
"C:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4808 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4808 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\cores\\coreX-corefiles" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\variants\\32pin-standard" "-Ic:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src" "-IC:\\Users\\user\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.3\\libraries\\SPI\\src" "c:\\Users\\user\\Documents\\Arduino\\libraries\\Ethernet\\src\\utility\\w5100.cpp" -o "C:\\Users\\user\\AppData\\Local\\Temp\\arduino\\sketches\\62E2E3EFDD73DEE05A85442B7B18B5D6\\libraries\\Ethernet\\utility\\w5100.cpp.o"
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp: In member function 'int DNSClient::getHostByName(const char*, arduino::IPAddress&, uint16_t)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp:99:20: error: 'INADDR_NONE' was not declared in this scope
  if (iDNSServer == INADDR_NONE) {
                    ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp:99:20: note: suggested alternative:
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp:5:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:78:24: note:   'arduino::INADDR_NONE'
 extern const IPAddress INADDR_NONE;
                        ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp: In member function 'uint16_t DNSClient::ProcessResponse(uint16_t, arduino::IPAddress&)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp:340:35: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
    iUdp.read(aAddress.raw_address(), 4);
                                   ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Dns.cpp:5:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\socket.cpp: In static member function 'static uint8_t EthernetClass::socketBeginMulticast(uint8_t, arduino::IPAddress, uint16_t)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\socket.cpp:185:42: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
      W5100.writeSnDIPR(s, ip.raw_address());   //239.255.0.1
                                          ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\socket.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static int EthernetClass::begin(uint8_t*, long unsigned int, long unsigned int)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:38:52: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setIPAddress(IPAddress(0,0,0,0).raw_address());
                                                    ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:47:54: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
   W5100.setIPAddress(_dhcp->getLocalIp().raw_address());
                                                      ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:48:56: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
   W5100.setGatewayIp(_dhcp->getGatewayIp().raw_address());
                                                        ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:49:58: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
   W5100.setSubnetMask(_dhcp->getSubnetMask().raw_address());
                                                          ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static void EthernetClass::begin(uint8_t*, arduino::IPAddress, arduino::IPAddress, arduino::IPAddress, arduino::IPAddress)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:86:36: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setIPAddress(ip.raw_address());
                                    ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:87:41: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setGatewayIp(gateway.raw_address());
                                         ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:88:41: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setSubnetMask(subnet.raw_address());
                                         ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static int EthernetClass::maintain()':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:132:55: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
    W5100.setIPAddress(_dhcp->getLocalIp().raw_address());
                                                       ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:133:57: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
    W5100.setGatewayIp(_dhcp->getGatewayIp().raw_address());
                                                         ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:134:59: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
    W5100.setSubnetMask(_dhcp->getSubnetMask().raw_address());
                                                           ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static arduino::IPAddress EthernetClass::localIP()':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:158:37: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.getIPAddress(ret.raw_address());
                                     ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static arduino::IPAddress EthernetClass::subnetMask()':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:167:38: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.getSubnetMask(ret.raw_address());
                                      ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In static member function 'static arduino::IPAddress EthernetClass::gatewayIP()':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:176:37: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.getGatewayIp(ret.raw_address());
                                     ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In member function 'void EthernetClass::setLocalIP(arduino::IPAddress)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:192:36: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setIPAddress(ip.raw_address());
                                    ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In member function 'void EthernetClass::setSubnetMask(arduino::IPAddress)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:200:37: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setSubnetMask(ip.raw_address());
                                     ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp: In member function 'void EthernetClass::setGatewayIP(arduino::IPAddress)':
c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:208:36: error: 'uint8_t* arduino::IPAddress::raw_address()' is private within this context
  W5100.setGatewayIp(ip.raw_address());
                                    ^
In file included from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\user\Documents\Arduino\libraries\Ethernet\src\Ethernet.cpp:21:
C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\cores\coreX-corefiles/api/IPAddress.h:42:12: note: declared private here
   uint8_t* raw_address() { return _address.bytes; };
            ^~~~~~~~~~~
Multiple libraries were found for "Ethernet.h"
  Used: C:\Users\user\Documents\Arduino\libraries\Ethernet
  Not used: C:\Users\user\AppData\Local\Arduino15\libraries\Ethernet
Using library Ethernet at version 2.0.2 in folder: C:\Users\user\Documents\Arduino\libraries\Ethernet 
Using library SPI at version 1.0 in folder: C:\Users\user\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.3\libraries\SPI 
exit status 1

Compilation error: exit status 1
```

--- a/megaavr/cores/coreX-corefiles/api/IPAddress.h
+++ b/megaavr/cores/coreX-corefiles/api/IPAddress.h
@@ -24,6 +24,10 @@
 #include "Printable.h"
 #include "String.h"
 
+class EthernetClass;
+class DhcpClass;
+class DNSClient;
+
 namespace arduino {
 // A class to make it easier to handle and pass around IP addresses
 
@@ -67,12 +71,13 @@ class IPAddress : public Printable
 
   virtual size_t printTo(Print& p) const;
 
-  friend class EthernetClass;
   friend class UDP;
   friend class Client;
   friend class Server;
-  friend class DhcpClass;
-  friend class DNSClient;
+
+  friend ::EthernetClass;
+  friend ::DhcpClass;
+  friend ::DNSClient;
 };
 
 extern const IPAddress INADDR_NONE;

